webrender/src/prim_store.rs

```diff
@@ -180,6 +181,24 @@ impl ToGpuBlocks for RectanglePrimitive {
     }
 }
 
+#[derive(Debug, Clone)]
+#[repr(C)]
+pub struct LinePrimitive {
+    pub color: ColorF,
+    pub style: LineStyle,
+    pub orientation: LineOrientation,
+}
+
+impl ToGpuBlocks for LinePrimitive {
+    fn write_gpu_blocks(&self, mut request: GpuDataRequest) {
+        request.push(self.color);
+        request.push([pack_as_float(self.style as u32),
```

Comment:
what's the reason to call `pack_as_float` here as opposed to just casting to `f32`?